actors/ppo.py

```diff
@@ -0,0 +1,720 @@
+# Copyright 2022 AI Redefined Inc. <dev+cogment@ai-r.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from dataclasses import dataclass
+from typing import Union, Tuple, List
+
+import cogment
+import numpy as np
+import torch
+from torch.distributions.normal import Normal
+
+from cogment_verse import Model
+from cogment_verse.run.run_session import RunSession
+from cogment_verse.run.sample_producer_worker import SampleProducerSession
+from cogment_verse.specs import (
+    PLAYER_ACTOR_CLASS,
+    AgentConfig,
+    EnvironmentConfig,
+    EnvironmentSpecs,
+    PlayerAction,
+    cog_settings,
+    flatten,
+    flattened_dimensions,
+    unflatten,
+)
+
+log = logging.getLogger(__name__)
+
+# pylint: disable=E1102
+# pylint: disable=W0212
+class PolicyNetwork(torch.nn.Module):
+    """Gaussian policy network"""
+
+    def __init__(self, num_input: int, num_output: int, num_hidden: int) -> None:
+        super().__init__()
+        self.input = torch.nn.Linear(num_input, num_hidden)
```

Comment:
a recent implementation also recommends layer initialization: https://github.com/vwxyzjn/ppo-implementation-details/blob/main/ppo.py#L95

Reply:
Yes, it has been implemented in `ppo.py` (line 225-226) where the orthogonal initialization has been used